cbits/ryu.h

```diff
@@ -0,0 +1,4 @@
+
+#define F2S_MAX_DIGITS 16
+#define D2S_MAX_DIGITS 25
```

Comment:
Does the C code use these anywhere?  Just moving the magic to a C header does not entirely solve the problem.

Reply:
Not currently, since the buffers are provided to it. Would you prefer they be spelled out in terms of the components?

Reply:
I was looking for evidence in the C code that `to_chars()` will write at most that much output...  How do we know that `16` or `25` are the right limits?

Reply:
Good point... I'm not sure they easily extractible from the algorithm but they rely on the following (which I can put in a comment?)
1. The maximal number of decimal digits needed to encode an n-bit floating point (which is 9 and 17 for float and decimal respectively). These bounds are 'well known' but are derived from [In-and-Out conversions](https://dl.acm.org/citation.cfm?id=362887). which state the binary-decimal-binary round-trip can be satisfied for an `n` digit binary number in `ceil(n*(log(2)/log(10))) + 1`.
2. That Ryu is correct and produces the shortest representation (in general, but especially for max-length values)

The `decimalLength9` and `decimalLength17` functions also implicitly rely on these facts. The other digits come from punctuation (`-`, `.`, `e`, `-` again in exponent) and the exponent (-38<->+38 for floats and -308<->+308 for doubles)

Reply:
Updated https://github.com/haskell/bytestring/pull/222/commits/4db3078c8cd6caaf65ad50929bfa847f2643b432